### PR TITLE
Use fxhash and add `add_raw` to SourceMapBuilder

### PR DESCRIPTION
**Pull request by kdy1**
add_raw is very important for performance as `SourceMapBuilder#add_source` copies the whole source code.

### CODE REVIEW

File: src/builder.rs [hunk 1]
```diff
@@ -183,6 +183,36 @@ impl SourceMapBuilder {
...
+        &mut self,
+        dst_line: u32,
+        dst_col: u32,
+        src_line: u32,
+        src_col: u32,
+        source: Option<u32>,
+        name: Option<u32>,
+    ) -> RawToken {
+        let src_id = match source {
+            Some(source) => source,
```
Comment (Swatinem) [inline src/builder.rs]: Should this index be validated? The `!0` suggests we don’t really care?

Comment (kdy1) [reply src/builder.rs]: It's same as add.

https://github.com/getsentry/rust-sourcemap/blob/97cf8ebc536c7cc53e1f103086e8d0f150470ae6/src/builder.rs#L166-L172

**Comment by kdy1** [comment]
will you merge the pr if I remove fxhash dependency?

**Comment by Swatinem** [comment]
> will you merge the pr if I remove fxhash dependency?

I think we will, yes

**Pull request closed without merging**